### ISSUE TITLE
NMM 0.62.18 - NMM gets confused when auto-installing old and new mod files with the same Download ID when switching profiles

### ISSUE THREAD

**Issue by KunoMochi** [Nexus-Mods/NMM-Test-Builds#107]
**Issue:**
Uninstalled (but not remove) old mod file then downloaded a new version of a mod file. Both have the same Download ID (possibly from mod author updating the file download link instead of making a new download link for file).

Switched to a profile without the mod installed (i.e., an empty profile) then switched back, NMM will ask if you want to overwrite files from the old version with the new version.

[_Note_]: If the old mod version is uninstalled again from the profile then switched profiles again, the new version will still be seen as installed with an "Install Date" but also a "Deactivated" icon. (See images below)

**Steps to Reproduce:**
1. Install an outdated mod file (if you have none, make a duplicate of the mod archive file then manually retrieve the mod info for the current version and type in an older version number)
2. Download new version of mod file
3. Uninstall old version of mod file
4. Install new version of mod file
5. Switch to a profile that does not have that mod file installed
6. Switch back to the profile with the mod file installed

**Images:**
![nmm0-62-18_profileswitchautoinstall1](https://cloud.githubusercontent.com/assets/7691651/16055251/94abfd8c-323e-11e6-8cdd-7a70225763a5.jpg)
![nmm0-62-18_profileswitchautoinstall2](https://cloud.githubusercontent.com/assets/7691651/16055295/aa87e8c8-323e-11e6-8866-6b4bb206d462.jpg)
[_Images for Note section_]
![nmm0-62-18_profileswitchautoinstall3](https://cloud.githubusercontent.com/assets/7691651/16055300/ad4df39a-323e-11e6-8fd5-e6f8a97959be.jpg)
![nmm0-62-18_namedoubledash](https://cloud.githubusercontent.com/assets/7691651/16050490/d89aa5b2-3229-11e6-988e-5a5f5634ddae.jpg)
![nmm0-62-18_profileswitchautoinstall4](https://cloud.githubusercontent.com/assets/7691651/16055304/b29716d8-323e-11e6-87ce-8bf7e616d909.jpg)

**Comment by DuskDweller** [comment]
Can you please go in the Mods folder for Skyrim and check the filename of the 2.8 version of Vivid Landscapes Moss Rocks ?

**Comment by KunoMochi** [comment]
**Old version (2.6) file name:** Vivid Landscapes - Moss Rocks - Compatibility Patch-49344-2-6.7z
**New version (2.8) file name:** Vivid Landscapes - Moss Rocks - Compatibility Patch-49344-2-8.7z

**Comment by DuskDweller** [comment]
Do you still get this issue? We tried again and again with the same mods but we're unable to replicate it.

**Comment by KunoMochi** [comment]
Yeah, I still get the issue happening. Every time I switch between profiles, it would install both versions with the same Download ID.

If you are having troubles getting files with the same Download ID to test with, I made a step-by-step instructions with images:
![nmm0-62-19_downloadidconfusion01](https://cloud.githubusercontent.com/assets/7691651/16316922/c40a9b00-3956-11e6-9eca-83871a49b4c1.png)
![nmm0-62-19_downloadidconfusion02](https://cloud.githubusercontent.com/assets/7691651/16316924/c5a6c51a-3956-11e6-9ad0-c57a4581b9b0.png)
![nmm0-62-19_downloadidconfusion03](https://cloud.githubusercontent.com/assets/7691651/16316925/c77fb9b4-3956-11e6-8da4-6410a845d3a7.png)
![nmm0-62-19_downloadidconfusion04](https://cloud.githubusercontent.com/assets/7691651/16316927/c9b1aad0-3956-11e6-8b75-9ea6f51232f9.png)
![nmm0-62-19_downloadidconfusion05](https://cloud.githubusercontent.com/assets/7691651/16316929/cb05e87e-3956-11e6-89d8-68a4c42d016d.png)
![nmm0-62-19_downloadidconfusion06](https://cloud.githubusercontent.com/assets/7691651/16316933/cd6b4f0a-3956-11e6-936a-afb172126a88.png)
